### PR TITLE
PD-1418: Updates Tooltip within Copyable

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Resolves Tooltip presentation issues when using a Copyable within a Modal

🎟 _Jira ticket:_ [PD-1418](https://jira.mongodb.org/browse/PD-1418)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

## 🧪 How to test changes

Use a Copyable within a Modal with `z-index > 0`:
Create a temporary Story within `Copyable.story.tsx` 
```js
  .add('Within Modal', () => {
    const darkMode = boolean('Dark mode', false)
    return (
      <Modal
        open={true}
        size='default'
        darkMode={darkMode}
        style={{ zIndex: 1 }}
      >
        <Copyable
          label={text('Label', 'Label')}
          description={text('Description', '') || undefined}
          darkMode={darkMode}
          shouldTooltipUsePortal={false}
        >
          {text('Text', 'npm install @leafygreen-ui/copyable')}
        </Copyable>
      </Modal>
    )
  })
```

### Before
Before

### After
After

### CODE REVIEW

File: .changeset/cyan-turkeys-run.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/copyable': patch
+---
+
+Fixes Copyable Tooltip when used within a Modal
```
Comment (brookescarlett) [inline .changeset/cyan-turkeys-run.md:5]: Let's be a little clearer in the changeset: 

something like: Tooltips that now appear in association with Copyable component are no longer portaled, in order to support usage within a Modal

File: packages/copyable/src/Copyable.tsx [hunk 2]
```diff
@@ -178,6 +178,7 @@ export default function Copyable({
         justify={Justify.Middle}
         trigger={trigger}
         triggerEvent={TriggerEvent.Click}
+        usePortal={false}
```
Comment (TheSonOfThomp) [inline packages/copyable/src/Copyable.tsx:181]: Are there any cases where the copyable tooltip _should_ use a portal?

Comment (brookescarlett) [reply packages/copyable/src/Copyable.tsx:181]: We might want to expose a prop here:

shouldPortalTooltip comes to mind but open to others, we can default to `false` as you have here.

Comment (brookescarlett) [reply packages/copyable/src/Copyable.tsx:181]: Actually, let's default to `true`, in which case this is only a minor change rather than a breaking one.

**Comment by brookescarlett** [comment]
Also, @TheSonOfThomp, don't forget to run `yarn changeset`

File: packages/copyable/src/Copyable.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/copyable/src/Copyable.tsx:181]: (that's also more in line with how we default components that consume our Portal component)

**Comment by brookescarlett** [comment]
We just need to document the new prop in the component's readme!

**Review by brookescarlett — APPROVED** [review]

**Pull request merged**